src/VirtoCommerce.ImageToolsModule.Data/Services/DefaultImageService.cs

```diff
@@ -1,26 +1,63 @@
 using System;
 using System.IO;
+using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 using SixLabors.ImageSharp;
 using SixLabors.ImageSharp.Formats;
 using SixLabors.ImageSharp.Formats.Jpeg;
 using SixLabors.ImageSharp.PixelFormats;
 using VirtoCommerce.AssetsModule.Core.Assets;
+using VirtoCommerce.ImageToolsModule.Core;
 using VirtoCommerce.ImageToolsModule.Core.Models;
 using VirtoCommerce.ImageToolsModule.Core.Services;
+using VirtoCommerce.Platform.Core.Settings;
 
 namespace VirtoCommerce.ImageToolsModule.Data.Services
 {
     public class DefaultImageService : IImageService
     {
         private readonly IBlobStorageProvider _storageProvider;
         private readonly ILogger<DefaultImageService> _logger;
+        private readonly ISettingsManager _settingsManager;
+        private IImageFormat[] _allowedImageFormats;
 
-        public DefaultImageService(IBlobStorageProvider storageProvider, ILogger<DefaultImageService> logger)
+        public DefaultImageService(IBlobStorageProvider storageProvider, ISettingsManager settingsManager, ILogger<DefaultImageService> logger)
         {
             _storageProvider = storageProvider;
             _logger = logger;
+            _settingsManager = settingsManager;
+        }
+
+        public virtual async Task<bool> IsExtensionAllowed(string path)
+        {
+            var allowedImageFormats = await GetAllowedImageFormats();
+
+            var extension = Path.GetExtension(path).TrimStart('.');
+
+            return allowedImageFormats.SelectMany(x => x.FileExtensions).Contains(extension, StringComparer.OrdinalIgnoreCase);
+        }
+
+        public virtual async Task<bool> IsImageFormatAllowed(IImageFormat format)
+        {
+            var allowedImageFormats = await GetAllowedImageFormats();
+
+            return allowedImageFormats.Any(f => string.Equals(f.Name, format.Name, StringComparison.OrdinalIgnoreCase));
+        }
+
+        protected virtual async Task<IImageFormat[]> GetAllowedImageFormats()
```

Comment:
Looks like _allowedImageFormats will be filled more often than null. For better performance we may consider to use ValueTask instead of classic Task hear.